drivers/ethernet/eth_nxp_imx_netc.c

```diff
@@ -286,6 +293,12 @@ enum ethernet_hw_caps netc_eth_get_capabilities(const struct device *dev)
 		| ETHERNET_PROMISC_MODE
 #endif
 	);
+
+	if (cfg->pseudo_mac) {
```

Comment:
Just making sure I understand this- will all instances of this IP with an internal PHY support DSA? Or are these functions unrelated?

Reply:
This eth_nxp_imx_netc.c driver is for normal ENETC MAC of NETC. There may be several instances designed for external usage. For example, i.MX95, there are 3 ENETC MACs externally but without switch IP in NETC.

For SoC like i.MX RT1180, there is also 5 ports switch IP in NETC together with 2 ENETC MACs. So, usually one ENETC MAC will connect to one switch MAC inside SoC for DSA management. The MACs connected inside SoC is called pseudo MAC in reference manual. Outside the SoC, we could see 4 switch ports and one normal eth port.
Thanks.

![image](https://github.com/user-attachments/assets/9e918e73-2a16-45ed-ab8c-e97fbb3dc026)